### ISSUE TITLE
Reorganize the API page by topic

### ISSUE THREAD

**Issue by matthewp** [stealjs/stealjs#76]
__tldr;__ Our API docs should be grouped by the common topics that people need help with. This proposal seeks to update the API page to make the important topics stand out.

> This was discussed at a [contributors meeting (58:12)](https://youtu.be/95OX6-qii24?t=58m12s)

# The Problem

The API docs page looks like this:

![screen shot 2017-09-22 at 7 05 14 am](https://user-images.githubusercontent.com/361671/30741842-8a9ff714-9f64-11e7-89b4-453c5b508f03.png)

And this isn't even all of it! The problems are:

* *Guides* and *Topics* are really kind of the same thing. In this case guides are step-by-step instructions, but there is little overlap, so it doesn't make sense to separate them.
* Common things people care about are missing, like live-reload.

# Solution

I would like to group our docs by the __topic__ they are associated with. This will make it easier for users to find exactly what they are looking for.

It might look like this:

* Development
  * Getting Started
  * Progressive Loading
  * Development Bundles
  * Using NPM Packages
  * Hot Module Swapping
  * Configuration
  * Shimming
  * Conditional Loading
  * Using Babel
  * Circular Dependencies
* Bundling
  * Optimized Builds
  * Tree Shaking
  * Code Splitting
  * Multi-main Builds
  * Project Exporting
* Production
  * Loading Scripts from a CDN
  * Bundle Manifests
* Advanced
  * Writing Plugins
  * Writing Extensions

# Implementation

## Design

Each link to these topics should take the user to a __portal__. The portal is a place to:

* Learn from a high level what the topic is about and how Steal helps.
* Link to API documentation (steal and steal-tools, even ecosystem packages).
* Link to any step-by-step guides.

# Tasks

* [x] - Finalize the list of topics.
* [ ] - Move existing topics out of steal/steal-tools and into stealjs repo.
* [ ] - Create new groupings for Development, Bundling, Advanced.
* [ ] - Write new docs that we don't currently have.

**Comment by matthewp** [comment]
# WIP

* Development
  * [ ] - Getting Started
  * [ ] - Progressive Loading
  * [ ] - Development Bundles
  * [ ] - Using NPM Packages
  * [ ] - Hot Module Swapping
  * [ ] - Configuration
  * [ ] - Shimming
  * [ ] - Conditional Loading
  * [ ] - Using Babel
  * [ ] - Circular Dependencies
* Bundling
  * [ ] - Optimized Builds
  * [ ] - Code Splitting
  * [ ] - Multi-main Builds
  * [ ] - Project Exporting
* Production
  * [ ] - Loading Scripts from a CDN
  * [ ] - Bundle Manifests
* Advanced
  * [ ] - Writing Plugins
  * [ ] - Writing Extensions

**Comment by matthewp** [comment]
Done!
